client/my-sites/post-relative-time-status/index.jsx

```diff
@@ -36,14 +37,36 @@ class PostRelativeTime extends React.PureComponent {
 			case 'new':
 				return null;
 			case 'draft':
+			case 'future':
 			case 'pending':
 				return this.props.post.modified;
 			default:
 				return this.props.post.date;
 		}
 	}
 
-	getRelativeTimeText() {
+	getDisplayedTimeFromPost( post ) {
+		const moment = this.props.moment;
+
+		const now = moment();
+
+		if ( ! post ) {
+			// Placeholder text: "a few seconds ago" in English locale
+			return now.fromNow();
+		}
+
+		const { status, modified, date } = post;
+		const time = moment( includes( [ 'draft', 'pending', 'future' ], status ) ? modified : date );
+		if ( now.diff( time, 'days' ) >= 7 ) {
+			// Like "Mar 15, 2013 6:23 PM" in English locale
+			return time.format( 'lll' );
+		}
+
+		// Like "3 days ago" in English locale
+		return time.fromNow();
```

Comment:
I've not looked too deeply into this, nor have I tested it, but I wonder if the same logic from #40779 could be applied here?

```suggestion
		switch ( status ) {
			// Display relative time for published posts: 9 hours ago
			case 'publish':
				return time.fromNow();

			// Display the time, day and date for scheduled posts: Wed, 21 Apr 2020 14:00
			case 'future':
				return time.format( 'llll' );

			// Display the date and time for other posts: 21 Apr 2020 14:00
			default:
				return time.format( 'lll' );
		}
```